yfinance/base.py

```diff
@@ -533,7 +533,7 @@ def _fix_unit_mixups(self, df, interval, tz_exchange):
         return df
 
 
-    def _get_ticker_tz(self):
+    def _get_ticker_tz(self, proxy=None, timeout=None):
```

Comment:
Is _get_ticker_tz called without proxy and timeout in any other place? 
As proxy and timeout is passed to the history method and not Ticker constructor, any calls to this from another methods might not have the correct proxy. (However I do not fully understand the use of proxies in yfinance so this might not be an issue).
Also if timeout is None the request will hang indefinitely, so maybe default to something reasonable.
https://requests.readthedocs.io/en/latest/user/quickstart/#timeouts
Maybe to make sure any new code do not call it without setting these do not make them optional and fore timeout to something else than None.